packages/calling-stateful-client/src/DeviceManagerDeclarative.ts

```diff
@@ -141,6 +141,32 @@ class ProxyDeviceManager implements ProxyHandler<DeviceManager> {
   }
 }
 
+/** Temporary helper function to dedupe duplicate video devices obtained from SDK */
+const dedupeVideoDevices = (devices: VideoDeviceInfo[]): VideoDeviceInfo[] => {
```

Comment:
nit: use generics to avoid duplicate code, e.g.
```typescript
const dedupe = <T extends VideoDeviceInfo | AudioDeviceInfo>(devices: T[]): T[] => {
  const ids = new Set();
  const uniqueDevices: T[] = [];
  devices.forEach((device: T) => {
    if (!ids.has(device.id)) {
      uniqueDevices.push(device);
      ids.add(device.id);
    }
  });
  return uniqueDevices;
};
```

Reply:
+1
I'd go further and define an anonymous interface that tells me exactly what I need to use the generic function:

```ts
const dedupe = <T extends {id: string}>(devices: T[]): T[] => {
  const ids = new Set();
  const uniqueDevices: T[] = [];
  devices.forEach((device: T) => {
    if (!ids.has(device.id)) {
      uniqueDevices.push(device);
      ids.add(device.id);
    }
  });
  return uniqueDevices;
};
```

Reply:
Thanks guys! I was trying to use generics to dedupe my dedupers by looking online but did not quite understand. Your suggestions really helped :)